File: lead-validation-international-nodejs-examples/validate_lead_international_rest_example.js
import { ValidateLeadInternationalClient } from "../lead-validation-international-nodejs/src/REST/validate_lead_international_rest.js";

export async function validateLeadInternationalGo(licenseKey, isLive) {
    console.log("\n-----------------------------------------------------------------------");
    console.log("LeadValidationInternational - ValidateLeadInternationalInput - REST SDK");
    console.log( "-----------------------------------------------------------------------");

    const full_name = "Tim Cook";
    const salutation = "";
    const first_name = "Tim";
    const last_name = "Cook";
    const business_name = "Apple";
    const business_domain = "apple.com";
    const business_EIN = "";
    const address_line1 = "1 Infinite Loop";
    const address_line2 = "";
    const address_line3 = "";
    const address_line4 = "";
    const address_line5 = "";
    const locality = "Cupertino";
    const admin_area = "CA";
    const postal_code = "95014-2083";
    const country = "USA";
    const phone1 = "1-408-996-1010";
    const phone2 = "";
    const email = "tim@apple.com";
    const ip_address = "";
    const gender = "M";
    const date_of_birth = "";
    const utc_capture_time = "2025-08-01T12:00:00Z";
    const output_language = "English";
    const test_type = "business-noip";
    const timeout_seconds = 15;

    console.log("\n* Input *\n");
    console.log(`Full Name        : ${full_name}`);
    console.log(`Salutation       : ${salutation}`);
    console.log(`First Name       : ${first_name}`);
    console.log(`Last Name        : ${last_name}`);
    console.log(`Business Name    : ${business_name}`);
    console.log(`Business Domain  : ${business_domain}`);
    console.log(`Business EIN     : ${business_EIN}`);
    console.log(`Address Line 1   : ${address_line1}`);
    console.log(`Address Line 2   : ${address_line2}`);
    console.log(`Address Line 3   : ${address_line3}`);
    console.log(`Address Line 4   : ${address_line4}`);
    console.log(`Address Line 5   : ${address_line5}`);
    console.log(`Locality         : ${locality}`);
    console.log(`Admin Area       : ${admin_area}`);
    console.log(`Postal Code      : ${postal_code}`);
    console.log(`Country          : ${country}`);
    console.log(`Phone 1          : ${phone1}`);
    console.log(`Phone 2          : ${phone2}`);
    console.log(`Email            : ${email}`);
    console.log(`IP Address       : ${ip_address}`);
    console.log(`Gender           : ${gender}`);
    console.log(`Date of Birth    : ${date_of_birth}`);
    console.log(`UTC Capture Time : ${utc_capture_time}`);
    console.log(`Output Language  : ${output_language}`);
    console.log(`Test Type        : ${test_type}`);
    console.log(`Timeout Seconds  : ${timeout_seconds}`);
    console.log(`Is Live Mode     : ${isLive}`);
    console.log(`License Key      : ${licenseKey}`);

    try {
        const response = await ValidateLeadInternationalClient.invoke(full_name,
                                                                    salutation,
                                                                    first_name,
                                                                    last_name,
                                                                    business_name,
                                                                    business_domain,
                                                                    business_EIN,
                                                                    address_line1,
                                                                    address_line2,
                                                                    address_line3,
                                                                    address_line4,
                                                                    address_line5,
                                                                    locality,
                                                                    admin_area,
                                                                    postal_code,
                                                                    country,
                                                                    phone1,
                                                                    phone2,
                                                                    email,
                                                                    ip_address,
                                                                    gender,
                                                                    date_of_birth,
                                                                    utc_capture_time,
                                                                    output_language,
                                                                    test_type,
                                                                    licenseKey,
                                                                    isLive,
                                                                    timeout_seconds
        );

        if (response.Error)
            return console.error("Error invoking ValidateLeadInternational:", response.Error);

        if (!response.Error) {
            console.log("\n* Lead Details *\n");
            console.log(`Overall Certainty     : ${response.OverallCertainty}`);
            console.log(`Overall Quality       : ${response.OverallQuality}`);
            console.log(`Lead Type             : ${response.LeadType}`);
            console.log(`Lead Country          : ${response.LeadCountry}`);
            console.log(`Note Codes            : ${response.NoteCodes}`);
            console.log(`Note Descriptions     : ${response.NoteDesc}`);

            console.log(`\n* Name Details *\n`);
            console.log(`Name Certainty        : ${response.NameCertainty}`);
            console.log(`Name Quality          : ${response.NameQuality}`);
            console.log(`First Name            : ${response.FirstName}`);
            console.log(`Last Name             : ${response.LastName}`);
            console.log(`First Name Latin      : ${response.FirstNameLatin}`);
            console.log(`Last Name Latin       : ${response.LastNameLatin}`);
            console.log(`Name Note Codes       : ${response.NameNoteCodes}`);
            console.log(`Name Note Descriptions: ${response.NameNoteDesc}`);

            console.log(`\n* Address Details *\n`);
            console.log(`Address Certainty     : ${response.AddressCertainty}`);
            console.log(`Address Quality       : ${response.AddressQuality}`);
            console.log(`Address Line 1        : ${response.AddressLine1}`);
            console.log(`Address Line 2        : ${response.AddressLine2}`);
            console.log(`Address Line 3        : ${response.AddressLine3}`);
            console.log(`Address Line 4        : ${response.AddressLine4}`);
            console.log(`Address Line 5        : ${response.AddressLine5}`);
            console.log(`Address Locality      : ${response.AddressLocality}`);
            console.log(`Address Admin Area    : ${response.AddressAdminArea}`);
            console.log(`Address Postal Code   : ${response.AddressPostalCode}`);
            console.log(`Address Country       : ${response.AddressCountry}`);
            console.log(`Address Resolution    : ${response.AddressResolutionLevel}`);
            console.log(`Address Note Codes    : ${response.AddressNoteCodes}`);
            console.log(`Address Note Desc     : ${response.AddressNoteDesc}`);

            console.log(`\n* Email Details *\n`);
            console.log(`Email Certainty       : ${response.EmailCertainty}`);
            console.log(`Email Quality         : ${response.EmailQuality}`);
            console.log(`Email Corrected       : ${response.EmailCorrected}`);
            console.log(`Email Note Codes      : ${response.EmailNoteCodes}`);
            console.log(`Email Note Desc       : ${response.EmailNoteDesc}`);

            console.log(`\n* IP Address Details *\n`);
            console.log(`IP Certainty          : ${response.IPCertainty}`);
            console.log(`IP Quality            : ${response.IPQuality}`);
            console.log(`IP Country            : ${response.IPCountry}`);
            console.log(`IP Locality           : ${response.IPLocality}`);
            console.log(`IP Admin Area         : ${response.IPAdminArea}`);
            console.log(`IP Note Codes         : ${response.IPNoteCodes}`);
            console.log(`IP Note Descriptions  : ${response.IPNoteDesc}`);

            console.log(`\n* Phone Details *\n`);
            console.log(`Phone 1 Certainty     : ${response.Phone1Certainty}`);
            console.log(`Phone 1 Quality       : ${response.Phone1Quality}`);
            console.log(`Phone 1 Locality      : ${response.Phone1Locality}`);
            console.log(`Phone 1 Admin Area    : ${response.Phone1AdminArea}`);
            console.log(`Phone 1 Country       : ${response.Phone1Country}`);
            console.log(`Phone 1 Note Codes    : ${response.Phone1NoteCodes}`);
            console.log(`Phone 1 Note Desc     : ${response.Phone1NoteDesc}`);

            console.log(`Phone 2 Certainty     : ${response.Phone2Certainty}`);
            console.log(`Phone 2 Quality       : ${response.Phone2Quality}`);
            console.log(`Phone 2 Locality      : ${response.Phone2Locality}`);
            console.log(`Phone 2 Admin Area    : ${response.Phone2AdminArea}`);
            console.log(`Phone 2 Country       : ${response.Phone2Country}`);
            console.log(`Phone 2 Note Codes    : ${response.Phone2NoteCodes}`);
            console.log(`Phone 2 Note Desc     : ${response.Phone2NoteDesc}`);

            console.log(`\n* Business Details *\n`);
            console.log(`Business Certainty    : ${response.BusinessCertainty}`);
            console.log(`Business Quality      : ${response.BusinessQuality}`);
            console.log(`Business Name         : ${response.BusinessName}`);
            console.log(`Business Domain       : ${response.BusinessDomain}`);
            console.log(`Business Email        : ${response.BusinessEmail}`);
            console.log(`Business Note Codes   : ${response.BusinessNoteCodes}`);
            console.log(`Business Note Desc    : ${response.BusinessNoteDesc}`);

            console.log("\n* Phone Contact *\n");
            if (response.PhoneContact) {
                console.log(`Name   : ${response.PhoneContact.Name}`);
                console.log(`Address: ${response.PhoneContact.Address}`);
                console.log(`City   : ${response.PhoneContact.City}`);
                console.log(`State  : ${response.PhoneContact.State}`);
                console.log(`Zip    : ${response.PhoneContact.Zip}`);
                console.log(`Type   : ${response.PhoneContact.Type}`);
            } else {
                console.log("No phone contact found.");
            }

            console.log("\n* Information Components *\n");
            if (response.InformationComponents && response.InformationComponents>0) {
                response.InformationComponents.forEach((component) => {
                    console.log(`Name : ${component.Name}`);
                    console.log(`Value: ${component.Value}\n`);
                });
            } else {
                console.log("No information components found.");
            }
        } else {
            console.log("\n* Error *\n");
            console.log(`Error Type       : ${response.Error.Type}`);
            console.log(`Error Type Code  : ${response.Error.TypeCode}`);
            console.log(`Error Description: ${response.Error.Desc}`);
            console.log(`Error Desc Code  : ${response.Error.DescCode}`);
        }
    } catch (e) {
        console.log(`\nException occurred: ${e.message}`);
    }
}

File: lead-validation-international-python-examples/validate_lead_international_rest_example.py
import sys
import os

sys.path.insert(0, os.path.abspath("../lead-validation-international-python/REST"))

from validate_lead_international_rest import validate_lead_international

def validate_lead_international_rest_sdk_go(license_key: str, is_live: bool) -> None:
    print("\n-----------------------------------------------------------------------")
    print("LeadValidationInternational - ValidateLeadInternationalInput - REST SDK")
    print("-----------------------------------------------------------------------")

    full_name = "Tim Cook"
    salutation = ""
    first_name = "Tim"
    last_name = "Cook"
    business_name = "Apple"
    business_domain = "apple.com"
    business_EIN = ""
    address_line1 = "1 Infinite Loop"
    address_line2 = ""
    address_line3 = ""
    address_line4 = ""
    address_line5 = ""
    locality = "Cupertino"
    admin_area = "CA"
    postal_code = "95014-2083"
    country = "USA"
    phone1 = "1-408-996-1010"
    phone2 = ""
    email = "tim@apple.com"
    ip_address = ""
    gender = "M"
    date_of_birth = ""
    utc_capture_time = "2025-08-01T12:00:00Z"
    output_language = "English"
    test_type = "business-noip"
    timeout_seconds = 15

    print("\n* Input *\n")
    print(f"Full Name        : {full_name}")
    print(f"Salutation       : {salutation}")
    print(f"First Name       : {first_name}")
    print(f"Last Name        : {last_name}")
    print(f"Business Name    : {business_name}")
    print(f"Business Domain  : {business_domain}")
    print(f"Business EIN     : {business_EIN}")
    print(f"Address Line 1   : {address_line1}")
    print(f"Address Line 2   : {address_line2}")
    print(f"Address Line 3   : {address_line3}")
    print(f"Address Line 4   : {address_line4}")
    print(f"Address Line 5   : {address_line5}")
    print(f"Locality         : {locality}")
    print(f"Admin Area       : {admin_area}")
    print(f"Postal Code      : {postal_code}")
    print(f"Country          : {country}")
    print(f"Phone 1          : {phone1}")
    print(f"Phone 2          : {phone2}")
    print(f"Email            : {email}")
    print(f"IP Address       : {ip_address}")
    print(f"Gender           : {gender}")
    print(f"Date of Birth    : {date_of_birth}")
    print(f"UTC Capture Time : {utc_capture_time}")
    print(f"Output Language  : {output_language}")
    print(f"Test Type        : {test_type}")
    print(f"Timeout Seconds  : {timeout_seconds}")
    print(f"Is Live Mode     : {is_live}")

    try:
        response = validate_lead_international(
            full_name, salutation, first_name, last_name,
            business_name, business_domain, business_EIN,
            address_line1, address_line2, address_line3,
            address_line4, address_line5,
            locality, admin_area, postal_code, country,
            phone1, phone2, email, ip_address,
            gender, date_of_birth, utc_capture_time,
            output_language, test_type, license_key,
            is_live, timeout_seconds
        )

        if response.Error is None:
            print("\n* Lead Details *\n")
            print(f"Overall Certainty     : {response.OverallCertainty}")
            print(f"Overall Quality       : {response.OverallQuality}")
            print(f"Lead Type             : {response.LeadType}")
            print(f"Lead Country          : {response.LeadCountry}")
            print(f"Note Codes            : {response.NoteCodes}")
            print(f"Note Descriptions     : {response.NoteDesc}")

            print("\n*  Name Details *\n")
            print(f"Name Certainty        : {response.NameCertainty}")
            print(f"Name Quality          : {response.NameQuality}")
            print(f"First Name            : {response.FirstName}")
            print(f"Last Name             : {response.LastName}")
            print(f"First Name Latin      : {response.FirstNameLatin}")
            print(f"Last Name Latin       : {response.LastNameLatin}")
            print(f"Name Note Codes       : {response.NameNoteCodes}")
            print(f"Name Note Descriptions: {response.NameNoteDesc}")

            print("\n*  Address Details *\n")
            print(f"Address Certainty     : {response.AddressCertainty}")
            print(f"Address Quality       : {response.AddressQuality}")
            print(f"Address Line 1        : {response.AddressLine1}")
            print(f"Address Line 2        : {response.AddressLine2}")
            print(f"Address Line 3        : {response.AddressLine3}")
            print(f"Address Line 4        : {response.AddressLine4}")
            print(f"Address Line 5        : {response.AddressLine5}")
            print(f"Address Locality      : {response.AddressLocality}")
            print(f"Address Admin Area    : {response.AddressAdminArea}")
            print(f"Address Postal Code   : {response.AddressPostalCode}")
            print(f"Address Country       : {response.AddressCountry}")
            print(f"Address Resolution    : {response.AddressResolutionLevel}")
            print(f"Address Note Codes    : {response.AddressNoteCodes}")
            print(f"Address Note Desc     : {response.AddressNoteDesc}")

            print("\n*  Email Details *\n")
            print(f"Email Certainty       : {response.EmailCertainty}")
            print(f"Email Quality         : {response.EmailQuality}")
            print(f"Email Corrected       : {response.EmailCorrected}")
            print(f"Email Note Codes      : {response.EmailNoteCodes}")
            print(f"Email Note Desc       : {response.EmailNoteDesc}")

            print("\n*  IP Address Details *\n")
            print(f"IP Certainty          : {response.IPCertainty}")
            print(f"IP Quality            : {response.IPQuality}")
            print(f"IP Country            : {response.IPCountry}")
            print(f"IP Locality           : {response.IPLocality}")
            print(f"IP Admin Area         : {response.IPAdminArea}")
            print(f"IP Note Codes         : {response.IPNoteCodes}")
            print(f"IP Note Descriptions  : {response.IPNoteDesc}")

            print("\n*  Phone Details *\n")
            print(f"Phone 1 Certainty     : {response.Phone1Certainty}")
            print(f"Phone 1 Quality       : {response.Phone1Quality}")
            print(f"Phone 1 Locality      : {response.Phone1Locality}")
            print(f"Phone 1 Admin Area    : {response.Phone1AdminArea}")
            print(f"Phone 1 Country       : {response.Phone1Country}")
            print(f"Phone 1 Note Codes    : {response.Phone1NoteCodes}")
            print(f"Phone 1 Note Desc     : {response.Phone1NoteDesc}")
            print(f"Phone 2 Certainty     : {response.Phone2Certainty}")

            print(f"Phone 2 Quality       : {response.Phone2Quality}")
            print(f"Phone 2 Locality      : {response.Phone2Locality}")
            print(f"Phone 2 Admin Area    : {response.Phone2AdminArea}")
            print(f"Phone 2 Country       : {response.Phone2Country}")
            print(f"Phone 2 Note Codes    : {response.Phone2NoteCodes}")
            print(f"Phone 2 Note Desc     : {response.Phone2NoteDesc}")

            print("\n* Business Details *\n")
            print(f"Business Certainty    : {response.BusinessCertainty}")
            print(f"Business Quality      : {response.BusinessQuality}")
            print(f"Business Name         : {response.BusinessName}")
            print(f"Business Domain       : {response.BusinessDomain}")
            print(f"Business Email        : {response.BusinessEmail}")
            print(f"Business Note Codes   : {response.BusinessNoteCodes}")
            print(f"Business Note Desc    : {response.BusinessNoteDesc}")

            print("\n* Phone Contact *\n")
            if hasattr(response, 'PhoneContact') and response.PhoneContact:
                print(f"Name   : {response.PhoneContact.Name}")
                print(f"Address: {response.PhoneContact.Address}")
                print(f"City   : {response.PhoneContact.City}")
                print(f"State  : {response.PhoneContact.State}")
                print(f"Zip    : {response.PhoneContact.Zip}")
                print(f"Type   : {response.PhoneContact.Type}")
            else:
                print("No phone contact returned.")

            print("\n* Information Components *\n")
            if hasattr(response, 'InformationComponents') and response.InformationComponents:
                for component in response.InformationComponents:
                    print(f"Name : {component.Name}")
                    print(f"Value: {component.Value}")
                    print("")
            else:
                print("No information components returned.")
        else:
          if hasattr(response, 'Error') and response.Error:
            print("\n* Error *\n")
            print(f"Error Type       : {response.Error.Type}")
            print(f"Error Type Code  : {response.Error.TypeCode}")
            print(f"Error Description: {response.Error.Desc}")
            print(f"Error Desc Code  : {response.Error.DescCode}")

    except Exception as e:
        print(f"\nException occurred: {str(e)}")

